### ISSUE TITLE
Reproducing Table 4 PSNR for Knee MRIs (MRNet) with 2D MedVAE

### ISSUE THREAD

**Issue by sebasmos** [StanfordMIMI/MedVAE#33]
<p>Hi, thanks for the great work and for open sourcing the models.</p>
<p>I am trying to replicate the reconstruction quality results from Table 4 for knee MRIs using the 2D MedVAE models (<code>medvae_4_1_2d</code> and <code>medvae_4_3_2d</code>) on MRNet data. I consistently get ~3 dB lower PSNR than reported:</p>

Model | Our PSNR | Paper PSNR
-- | -- | --
f=16, C=1 | ~24.0 | 27.38
f=16, C=3 | ~27.7 | 31.52

<p>I have ruled out several possible causes: per slice vs volume level PSNR computation, center cropping to 160x160, ScaleIntensity normalization, posterior.sample vs posterior.mode, and slice selection (middle only vs all 56 slices). The gap is consistent across both models.</p>
<p>Could you clarify the exact evaluation pipeline used for Table 4 on knee MRIs? Specifically:</p>
<ol>
<li>What input resolution were the MRNet slices fed at (native 256x256 or upsampled)?</li>
<li>Were the HuggingFace checkpoint weights the same ones used to generate Table 4?</li>
<li>Is there evaluation code available for reproducing Table 4?</li>
</ol>
<p>Thanks in advance for any guidance.</p>

**Comment by ashwinkumargb** [comment]
Hi @sebasmos, these are great points. I looked into this more this evening, but realized I no longer have the dataset on the server, so I’m unable to exactly reproduce what you’re seeing.

1. MRNet was loaded at its native resolution. Please see the data loader below for reference. We applied a 160³ center crop to the volumes.

2. The Hugging Face 2D weights with LoRA are the same weights used for Table 4.

3. I’ve attached the metrics script we used to compute the reported results.

It’s also important to note that we evaluated all baselines on the same subset with 100 random samples drawn from the validation set using a fixed random seed.

Below are the full metrics:

f = 16, C = 1
{“PSNR”: 27.380359649658203, “MSE”: 0.007940096780657768, “MS-SSIM”: 0.9903741478919983, “FID-Inception”: 31.077085494995117, “FID-CLIP”: 3.576171875, “Samples”: 100}

f = 16, C = 3
{“PSNR”: 31.516273498535156, “MSE”: 0.003305113408714533, “MS-SSIM”: 0.9974101185798645, “FID-Inception”: 7.659248352050781, “FID-CLIP”: 1.333984375, “Samples”: 100}

Metrics script: [compute_vae_metrics_3d.py](https://github.com/user-attachments/files/26109198/compute_vae_metrics_3d.py)

Data loader used for metric computation:

```python
"""
Run Inference on 3D volumes by padding it to the nearest multiple of 64 in all dimensions
@input: path to image (str)
@Output: padding (np.array)
"""
def load_mri_3d_metrics(path: str, dtype: torch.dtype = torch.float32, roi_size=[160, 160, 160], **kwargs):    
    # Mri transforms    
    mri_transforms = Compose(
        transforms=[LoadImage(),
        EnsureChannelFirst(),
        Orientation(axcodes="RAS"),
        ScaleIntensity(channel_wise=True, minv=0, maxv=1),
        MonaiNormalize(mean=[0.5], std=[0.5]),
        MonaiPad(size=roi_size, value=-1),
        CenterSpatialCrop(roi_size=roi_size),],
        lazy=True,
    )
    
    try: 
        mr_augmented = mri_transforms(path).as_tensor()
        return mr_augmented
    except Exception as e:
        print(f"Error in loading {path} with error: {e}")
        roi_size.insert(1)
        return torch.zeros(roi_size)
```
